src/components/AirdropFloatingButton.tsx

```diff
@@ -0,0 +1,55 @@
+import { StackNavigationProp } from '@react-navigation/stack';
+import React, { FC, useEffect } from 'react';
+import { StyleSheet, TouchableOpacity } from 'react-native';
+
+import { icons } from 'app/assets';
+import { Image } from 'app/components';
+import { Route } from 'app/consts';
+import { isAfterAirdrop } from 'app/helpers/airdrop';
+
+type Props = {
+  // thankYouSeen: boolean;
+  thankYouFlowCompleted: boolean;
+  navigation: StackNavigationProp<any, Route.Dashboard>;
+};
+
+export const AirdropFloatingButton: FC<Props> = ({ thankYouFlowCompleted, navigation }: Props) => {
+  // TODO: this implementation doesn't work.
+  // As discussed with Paweł, We have to merge PasswordNavigator with MainStackNavigator, otherwise it fails miserably.
+  // useEffect(() => {
+  //   if (navigation && !thankYouSeen) {
+  //     navigation.navigate(Route.AirdropThankYou);
+  //   }
+  // }, [navigation, thankYouSeen]);
+
+  const onButtonPress = () => {
+    if (isAfterAirdrop()) {
+      navigation.navigate(Route.AirdropDashboard);
```

Comment:
`return navigation.navigate(Route.AirdropDashboard);` because you fire navigation here and below

Reply:
done